trieste/acquisition/batch.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2022 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import functools
+
+import tensorflow as tf
+
+from ..types import TensorType
+from .interface import AcquisitionFunction
+
+
+def batchify_acquisition_function(
+    fn: AcquisitionFunction,
+    batch_size: int,
+) -> AcquisitionFunction:
+    """
+    A wrapper around an :const:`AcquisitionFunction` to split its input into batches.
+    Splits `x` into batches along the first dimension, calls `fn` on each batch, and then stitches
+    the results back together, so that it looks like `fn` was called with all of `x` in one batch.
+    :param fn: Acquisition function to call with batches of data.
+    :param batch_size: Call fn with tensors of at most this size.
+    :returns Batched acquisition function.
+    """
+    if batch_size <= 0:
+        raise ValueError(f"batch_size must be positive, got {batch_size}")
```

Comment:
I kinda think using the word batchify is confusing here as that means a very specific thing in our acq optimizers, i.e. batchify is to do with batches of query points. This PR is all about batching computations. Any thoughts on terminology @hstojic or @vpicheny ?

Reply:
Perhaps `throttled_acquisition_function`

Reply:
@vpicheny agreed that this is a bit confusing

Reply:
:+1: If there aren't any other suggestions I'll rename it to `throttle_acquisition_function` and `throttle_acquisition_function_calls`. Shall I continue to cal the batch size `batch_size` or shall I rename that to `throttle_batch_size`?

Reply:
Also any suggestions for the file name? It didn't obviously fit anywhere else. `throttle.py`? `memory.py`?

Reply:
maybe `memory_throttled` or `split` ? Any suggestions @hstojic or @vpicheny ?

Reply:
`hashed` or `sliced`? 

We should indeed NOT call anything else `batch_size`, this would be so confusing.

Reply:
yes, probably its best to avoid batch terms since it has very specific meaning in BO

I don't like throttle, but other than that I'm somewhat indifferent, I can add few more to the mix if you really need more :) `split_compute` and `split_optimize`

Reply:
batch_size -> split_size?